jsx/DynamicDataTable.js

```diff
@@ -1,88 +1,110 @@
 /* exported RDynamicDataTable */
 
-let DynamicDataTable = React.createClass({
-  propTypes: {
-    DataURL: React.PropTypes.string.isRequired,
-  },
+import React, {Component} from 'react';
+import PropTypes from 'prop-types';
+import StaticDataTable from 'StaticDataTable';
 
-  getInitialState: function() {
-    return {
+class DynamicDataTable extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
       Headers: [],
       Data: [],
       isLoaded: false,
       loadedData: 0,
     };
-  },
-  getDefaultProps: function() {
-    return {
-      DataURL: '',
-    };
-  },
-  componentDidMount: function() {
+
+    this.fetchData = this.fetchData.bind(this);
+    this.ajaxSuccess = this.ajaxSuccess.bind(this);
+    this.ajaxFailure = this.ajaxFailure.bind(this);
+    this.xhrFunction = this.xhrFunction.bind(this);
+    this.xhrProgress = this.xhrProgress.bind(this);
+  };
+
+  componentDidMount() {
     this.fetchData();
       // Listen for update event to update data table on outside changes
     window.addEventListener('update-datatable', this.fetchData);
-  },
-  componentWillUnmount: function() {
+  }
+
+  componentWillUnmount() {
       // Unsubscribe from the event before component is destroyed
     window.removeEventListener('update-datatable', this.fetchData);
-  },
-  fetchData: function() {
-    let that = this;
+  }
+
+  fetchData() {
     $.ajax(this.props.DataURL, {
       dataType: 'json',
       cache: false,
-      xhr: function() {
-        let xhr = new window.XMLHttpRequest();
-        xhr.addEventListener('progress', function(evt) {
-          that.setState({
-            loadedData: evt.loaded,
-          });
-        });
-        return xhr;
-      },
-      success: function(data) {
-        that.setState({
-          Headers: data.Headers,
-          Data: data.Data,
-          isLoaded: true,
-        });
-      },
-      error: function(data, errorCode, errorMsg) {
-        console.error(errorCode + ': ' + errorMsg);
-        that.setState({error: 'Error loading data'});
-      },
+      xhr: this.xhrFunction,
+      success: this.ajaxSuccess,
+      error: this.ajaxFailure,
     });
-  },
-  render: function() {
+  }
+
+  xhrFunction() {
+    let xhr = new window.XMLHttpRequest();
+    xhr.addEventListener('progress', this.xhrProgress);
+    return xhr;
+  }
+
+  xhrProgress(evt) {
+    this.setState({
+      loadedData: evt.loaded,
+    });
+  }
+
+  ajaxSuccess(data) {
+    this.setState({
+      Headers: data.Headers,
+      Data: data.Data,
+      isLoaded: true,
+    });
+  }
+
+  ajaxFailure(data, errorCode, errorMsg) {
+    console.error(errorCode + ': ' + errorMsg);
+    this.setState({error: 'Error loading data'});
```

Comment:
set state as 1 line or multiple lines like in xhrProgress() above?

Reply:
@zaliqarosli I'm not sure what you mean "set state as 1 line or multiple lines like in xhrProgress() above?" can you add to that?